<ui version="4.0" stdsetdef="1" >
  <author></author>
  <comment></comment>
  <exportmacro></exportmacro>
  <class>EncryptionConfigurationDialog</class>
  <widget class="QWidget" name="EncryptionConfigurationDialog" >
    <property name="geometry" >
      <rect>
        <x>0</x>
        <y>0</y>
        <width>445</width>
        <height>481</height>
      </rect>
    </property>
    <property name="windowTitle" >
      <string>Encryption Configuration</string>
    </property>
    <layout class="QVBoxLayout" >
      <property name="margin" >
        <number>11</number>
      </property>
      <property name="spacing" >
        <number>6</number>
      </property>
      <item>
        <widget class="KButtonGroup" name="encryptMessagesBG" >
          <property name="title" >
            <string>Encryption</string>
          </property>
          <layout class="QGridLayout" >
            <property name="margin" >
              <number>11</number>
            </property>
            <property name="spacing" >
              <number>6</number>
            </property>
            <item row="0" column="0" >
              <widget class="QLabel" name="encryptionAlgorithmLA" >
                <property name="text" >
                  <string>Encryption &amp;algorithm:</string>
                </property>
                <property name="buddy" stdset="0" >
                  <cstring>encryptionAlgorithmCO</cstring>
                </property>
                <property name="wordWrap" >
                  <bool>false</bool>
                </property>
              </widget>
            </item>
            <item rowspan="1" row="4" column="0" colspan="2" >
              <widget class="QCheckBox" name="storeEncryptedCB" >
                <property name="text" >
                  <string>Store sent messages &amp;encrypted</string>
                </property>
                <property name="checked" >
                  <bool>false</bool>
                </property>
                <property name="toolTip" stdset="0" >
                  <string>Check to store messages encrypted </string>
                </property>
                <property name="whatsThis" stdset="0" >
                  <string>&lt;qt>
&lt;h1>Store Messages Encrypted&lt;/h1>
When this box is checked, sent messages are stored encrypted like they were sent. This is not recommended, as you will not be able to read the messages any longer if a necessary certificate expires.
&lt;p>
However, there may be local rules that require you to turn this option on. When in doubt, check with your local administrator.
&lt;/qt></string>
                </property>
              </widget>
            </item>
            <item rowspan="1" row="1" column="0" colspan="2" >
              <widget class="QRadioButton" name="encryptAllPartsRB" >
                <property name="text" >
                  <string>Encr&amp;ypt all message parts</string>
                </property>
                <property name="checked" >
                  <bool>true</bool>
                </property>
                <property name="toolTip" stdset="0" >
                  <string>Check to have all message parts encrypted by default</string>
                </property>
                <property name="whatsThis" stdset="0" >
                  <string>&lt;qt>
&lt;h1>Encrypt All Message Parts By Default&lt;/h1>
If this option is selected, all parts of a message (i.e. the main message body and all attachments) are encrypted by default.
&lt;p>
This is a default setting, you can still override it for each individual message.
&lt;/qt></string>
                </property>
              </widget>
            </item>
            <item rowspan="1" row="2" column="0" colspan="2" >
              <widget class="QRadioButton" name="askEachPartRB" >
                <property name="text" >
                  <string>Ask &amp;before encrypting each part</string>
                </property>
                <property name="toolTip" stdset="0" >
                  <string>Check to be asked for each part whether to encrypt</string>
                </property>
                <property name="whatsThis" stdset="0" >
                  <string>&lt;qt>
&lt;h1>Ask Before Encrypting Each Part&lt;/h1>
When this option is selected, you will be asked for each part of the message (i.e. the main message body as well as all attachments) individually whether you want the part to be encrypted.
&lt;/qt></string>
                </property>
              </widget>
            </item>
            <item rowspan="1" row="3" column="0" colspan="2" >
              <widget class="QRadioButton" name="dontEncryptRB" >
                <property name="text" >
                  <string>&amp;Do not encrypt messages</string>
                </property>
                <property name="toolTip" stdset="0" >
                  <string>Check for not encrypting a message by default</string>
                </property>
                <property name="whatsThis" stdset="0" >
                  <string>&lt;qt>
&lt;h1>Do Not Encrypt Messages&lt;/h1>
If this option is selected, messages are not encrypted by default.
&lt;p>
This is a default setting, you can still override it for each individual setting.
&lt;/qt></string>
                </property>
              </widget>
            </item>
            <item row="0" column="1" >
              <widget class="QComboBox" name="encryptionAlgorithmCO" >
                <property name="toolTip" stdset="0" >
                  <string>Selects the encryption algorithm</string>
                </property>
                <property name="whatsThis" stdset="0" >
                  <string>&lt;qt>
&lt;h1>Encryption Algorithm&lt;/h1>
An algorithm is a description for the computer on how it should perform a certain task. The encryption algorithm describes how the computer applies your recipient's key to your message so that only the intended receiver can read the message.
&lt;p>
The selection of a certain encryption algorithm determines how easy or how difficult it is to intercept and read a message. However, all algorithms provided in the SPHINX environment are considered to be very safe. Generally, the default will work just fine here.
&lt;p>
This setting is a default, you can override it for each individual message.
&lt;/qt></string>
                </property>
                <item>
                  <property name="text" >
                    <string>Triple-DES</string>
                  </property>
                </item>
              </widget>
            </item>
            <item row="0" column="2" >
              <spacer name="Spacer19" >
                <property name="sizeHint" >
                  <size>
                    <width>20</width>
                    <height>20</height>
                  </size>
                </property>
                <property name="sizeType" >
                  <enum>Expanding</enum>
                </property>
                <property name="orientation" >
                  <enum>Horizontal</enum>
                </property>
              </spacer>
            </item>
          </layout>
        </widget>
      </item>
      <item>
        <widget class="KButtonGroup" name="certBG" >
          <property name="title" >
            <string>Certificates</string>
          </property>
          <layout class="QVBoxLayout" >
            <property name="margin" >
              <number>0</number>
            </property>
            <item>
              <widget class="KButtonGroup" name="certificatePathCheckBG" >
                <property name="title" >
                  <string/>
                </property>
                <layout class="QGridLayout" >
                  <property name="margin" >
                    <number>0</number>
                  </property>
                  <item rowspan="2" row="1" column="0" colspan="1" >
                    <spacer name="Spacer1_2_2" >
                      <property name="sizeHint" >
                        <size>
                          <width>20</width>
                          <height>20</height>
                        </size>
                      </property>
                      <property name="sizeType" >
                        <enum>Fixed</enum>
                      </property>
                      <property name="orientation" >
                        <enum>Horizontal</enum>
                      </property>
                    </spacer>
                  </item>
                  <item row="2" column="1" >
                    <widget class="QRadioButton" name="pathMayEndLocallyCB" >
                      <property name="text" >
                        <string>Check can end with locally sa&amp;ved certificate</string>
                      </property>
                      <property name="toolTip" stdset="0" >
                        <string>Check to end with locally saved certificate.</string>
                      </property>
                      <property name="whatsThis" stdset="0" >
                        <string>&lt;qt>
&lt;h1>Certificate Check Can End With Locally Saved Certificate&lt;/h1>
If this option is checked, the check of the certificate chain can end with a locally saved certificate.
&lt;p>
Locally saved certificates are your own certificates as well as the certificates of communication partners and certification authorities (CAs).
&lt;/h1></string>
                      </property>
                    </widget>
                  </item>
                  <item row="1" column="1" >
                    <widget class="QRadioButton" name="alwaysCheckRootRB" >
                      <property name="text" >
                        <string>Always c&amp;heck to root certificate</string>
                      </property>
                      <property name="checked" >
                        <bool>true</bool>
                      </property>
                      <property name="toolTip" stdset="0" >
                        <string>Check here to check up to the root certificate</string>
                      </property>
                      <property name="whatsThis" stdset="0" >
                        <string>&lt;qt>
&lt;h1>Always Check Certificate Path To Root Certificate&lt;/h1>
If this option is turned on, the certificate path belonging to the receiver's certificate will always be checked all the way to the root certificate.
&lt;/qt></string>
                      </property>
                    </widget>
                  </item>
                  <item rowspan="1" row="0" column="0" colspan="2" >
                    <widget class="QCheckBox" name="checkCertificatePathCB" >
                      <property name="text" >
                        <string>Check certificate &amp;path</string>
                      </property>
                      <property name="checked" >
                        <bool>true</bool>
                      </property>
                      <property name="toolTip" stdset="0" >
                        <string>Check here to have the whole certificate path checked</string>
                      </property>
                      <property name="whatsThis" stdset="0" >
                        <string>&lt;qt>
&lt;h1>Check Certificate Path&lt;/h1>
If this option is turned on, the whole path of the receiver's certificate up to the root will be checked.
&lt;p>
Note that it is not possible to turn off checking the receiver's certificate itself.
&lt;/qt></string>
                      </property>
                    </widget>
                  </item>
                </layout>
              </widget>
            </item>
            <item>
              <widget class="KButtonGroup" name="crlBG" >
                <property name="title" >
                  <string/>
                </property>
                <layout class="QGridLayout" >
                  <property name="margin" >
                    <number>0</number>
                  </property>
                  <item rowspan="1" row="0" column="0" colspan="2" >
                    <widget class="QCheckBox" name="useCRLsCB" >
                      <property name="text" >
                        <string>&amp;Use certificate revocation lists (CRLs)</string>
                      </property>
                      <property name="checked" >
                        <bool>true</bool>
                      </property>
                      <property name="toolTip" stdset="0" >
                        <string>Check to use CRLs</string>
                      </property>
                      <property name="whatsThis" stdset="0" >
                        <string>&lt;qt>
&lt;h1>Use Certificate Revocation Lists (CRLs)&lt;/h1>
A certificate revocation list contains certificates that have been withdrawn and should not be used for encryption purposes any longer. A user may wish to have his or her certificate revoked because he or she suspects that the certificate's integrity has been compromised (e.g. somebody has guessed the PIN).
&lt;p>
It is recommended to use CRLs for maximum security. In the configuration dialog for certificate and CRL management, you can select where to retrieve the CRLs from.
&lt;/qt></string>
                      </property>
                    </widget>
                  </item>
                  <item row="1" column="1" >
                    <widget class="QCheckBox" name="warnCRLExpireCB" >
                      <property name="text" >
                        <string>Warn if CRL e&amp;xpires in less than:</string>
                      </property>
                      <property name="checked" >
                        <bool>true</bool>
                      </property>
                      <property name="toolTip" stdset="0" >
                        <string>Check to be warned if a CRL expires in the near future</string>
                      </property>
                      <property name="whatsThis" stdset="0" >
                        <string>&lt;qt>
&lt;h1>Warn If CRL Expires In The Near Future&lt;/h1>
If this box is checked, you will be warned if one of the CRLs you are using is about to expire in the near future.
&lt;/qt></string>
                      </property>
                    </widget>
                  </item>
                  <item row="1" column="2" >
                    <widget class="QSpinBox" name="warnCRLExpireSB" >
                      <property name="suffix" >
                        <string> days</string>
                      </property>
                      <property name="maximum" >
                        <number>999</number>
                      </property>
                      <property name="value" >
                        <number>7</number>
                      </property>
                      <property name="toolTip" stdset="0" >
                        <string>Number of days before warning</string>
                      </property>
                      <property name="whatsThis" stdset="0" >
                        <string>&lt;qt>
&lt;h1>Warn If CRL Expires In The Near Future&lt;/h1>
In this field you can specify how soon before a CRL expires you are warned about this expiry.
&lt;p>
The recommended setting for the SPHINX environment is 7 days.
&lt;/qt></string>
                      </property>
                    </widget>
                  </item>
                  <item row="1" column="0" >
                    <spacer name="Spacer1_2" >
                      <property name="sizeHint" >
                        <size>
                          <width>20</width>
                          <height>20</height>
                        </size>
                      </property>
                      <property name="sizeType" >
                        <enum>Fixed</enum>
                      </property>
                      <property name="orientation" >
                        <enum>Horizontal</enum>
                      </property>
                    </spacer>
                  </item>
                  <item row="1" column="3" >
                    <spacer name="Spacer8" >
                      <property name="sizeHint" >
                        <size>
                          <width>20</width>
                          <height>20</height>
                        </size>
                      </property>
                      <property name="sizeType" >
                        <enum>Expanding</enum>
                      </property>
                      <property name="orientation" >
                        <enum>Horizontal</enum>
                      </property>
                    </spacer>
                  </item>
                </layout>
              </widget>
            </item>
            <item>
              <widget class="QCheckBox" name="alwaysEncryptToSelfCB" >
                <property name="text" >
                  <string>Always encrypt &amp;to self</string>
                </property>
                <property name="checked" >
                  <bool>true</bool>
                </property>
                <property name="toolTip" stdset="0" >
                  <string>Check to have encrypted messages also encrypted using your own key.</string>
                </property>
                <property name="whatsThis" stdset="0" >
                  <string>&lt;qt>
&lt;h1>Always encrypt to self&lt;/h1>
If this box is checked, encrypted messages sent by you will also be encrypted using your own key.
&lt;p>
It is recommended to leave this option turned on to enable you to read the messages you have sent.
&lt;/qt></string>
                </property>
              </widget>
            </item>
          </layout>
        </widget>
      </item>
      <item>
        <widget class="QGroupBox" name="encryptionSettingsBG" >
          <property name="title" >
            <string>Warnings</string>
          </property>
          <layout class="QGridLayout" >
            <property name="margin" >
              <number>11</number>
            </property>
            <property name="spacing" >
              <number>6</number>
            </property>
            <item rowspan="1" row="0" column="0" colspan="3" >
              <widget class="QCheckBox" name="warnUnencryptedCB" >
                <property name="text" >
                  <string>&amp;Warn when trying to send unencrypted messages</string>
                </property>
                <property name="checked" >
                  <bool>true</bool>
                </property>
                <property name="toolTip" stdset="0" >
                  <string>Check to be warned when sending unencrypted messages.</string>
                </property>
                <property name="whatsThis" stdset="0" >
                  <string>&lt;qt>
&lt;h1>Warn When Trying To Send Unencrypted Messages&lt;/h1>
If this box is checked, you will be warned when you try to send parts of or the whole message unencrypted.
&lt;p>
It is recommended to leave this option turned on for maximum integrity.
&lt;/qt></string>
                </property>
              </widget>
            </item>
            <item rowspan="1" row="4" column="1" colspan="2" >
              <widget class="QSpinBox" name="warnChainCertificateExpiresSB" >
                <property name="suffix" >
                  <string> days</string>
                </property>
                <property name="maximum" >
                  <number>999</number>
                </property>
                <property name="value" >
                  <number>14</number>
                </property>
                <property name="toolTip" stdset="0" >
                  <string>Select the number of days here</string>
                </property>
                <property name="whatsThis" stdset="0" >
                  <string>&lt;qt>
&lt;h1>Warn If A Certificate In The Chain Expires&lt;/h1>
Select the minimum number of days all certificates in the chain should be valid without issuing a warning.
&lt;p>
The recommended SPHINX setting is 14 days.
&lt;/qt></string>
                </property>
              </widget>
            </item>
            <item row="4" column="0" >
              <widget class="QCheckBox" name="warnChainCertificateExpiresCB" >
                <property name="text" >
                  <string>Warn if a certificate in &amp;the chain expires in less than:</string>
                </property>
                <property name="checked" >
                  <bool>true</bool>
                </property>
                <property name="toolTip" stdset="0" >
                  <string>Check to be warned if the certificate expires soon</string>
                </property>
                <property name="whatsThis" stdset="0" >
                  <string>&lt;qt>
&lt;h1>Warn If Certificate Expires&lt;/h1>
If this option is checked, then you will be warned when trying to use a certificate for encrypting that expires within the specified amount of days.
&lt;p>
It is recommended to keep this option turned on to avoid using certificates that expire in the near future.
&lt;/qt></string>
                </property>
              </widget>
            </item>
            <item rowspan="1" row="3" column="1" colspan="2" >
              <widget class="QSpinBox" name="warnReceiverCertificateExpiresSB" >
                <property name="suffix" >
                  <string> days</string>
                </property>
                <property name="maximum" >
                  <number>999</number>
                </property>
                <property name="value" >
                  <number>14</number>
                </property>
                <property name="toolTip" stdset="0" >
                  <string>Select the number of days here</string>
                </property>
                <property name="whatsThis" stdset="0" >
                  <string>&lt;qt>
&lt;h1>Warn If Encryption Certificate Expires&lt;/h1>
Select the minimum number of days the encryption certificate should be valid without issuing a warning.
&lt;p>
The recommended SPHINX setting is 14 days.
&lt;/qt></string>
                </property>
              </widget>
            </item>
            <item row="3" column="0" >
              <widget class="QCheckBox" name="warnReceiverCertificateExpiresCB" >
                <property name="text" >
                  <string>Warn if &amp;receiver certificate expires in less than:</string>
                </property>
                <property name="checked" >
                  <bool>true</bool>
                </property>
                <property name="toolTip" stdset="0" >
                  <string>Check to be warned if the certificate expires soon</string>
                </property>
                <property name="whatsThis" stdset="0" >
                  <string>&lt;qt>
&lt;h1>Warn If Certificate Expires&lt;/h1>
If this option is checked, then you will be warned when trying to use a certificate for encrypting that expires within the specified amount of days.
&lt;p>
It is recommended to keep this option turned on to avoid using certificates that expire in the near future.
&lt;/qt></string>
                </property>
              </widget>
            </item>
            <item rowspan="2" row="3" column="3" colspan="1" >
              <spacer name="Spacer7" >
                <property name="sizeHint" >
                  <size>
                    <width>20</width>
                    <height>20</height>
                  </size>
                </property>
                <property name="sizeType" >
                  <enum>Expanding</enum>
                </property>
                <property name="orientation" >
                  <enum>Horizontal</enum>
                </property>
              </spacer>
            </item>
            <item rowspan="2" row="1" column="0" colspan="2" >
              <widget class="QCheckBox" name="warnReceiverNotInCertificateCB" >
                <property name="text" >
                  <string>Warn if receiver's email address is &amp;not in certificate</string>
                </property>
                <property name="checked" >
                  <bool>true</bool>
                </property>
                <property name="toolTip" stdset="0" >
                  <string>Check to be warned if the address is not in the certificate</string>
                </property>
                <property name="whatsThis" stdset="0" >
                  <string>&lt;qt>
&lt;h1>Warn If Receiver's Email Address Is Not In Certificate&lt;/h1>
If this option is checked, a warning is issued if the email address of the receiver is not contained in the certificate used for encrypting.
&lt;p>
It is recommended to leave this option turned on for maximum security.
&lt;/qt></string>
                </property>
              </widget>
            </item>
          </layout>
        </widget>
      </item>
      <item>
        <spacer name="Spacer17" >
          <property name="sizeHint" >
            <size>
              <width>20</width>
              <height>20</height>
            </size>
          </property>
          <property name="sizeType" >
            <enum>Expanding</enum>
          </property>
          <property name="orientation" >
            <enum>Vertical</enum>
          </property>
        </spacer>
      </item>
    </layout>
  </widget>
  <layoutdefault spacing="6" margin="11" />
  <pixmapfunction>qPixmapFromMimeSource</pixmapfunction>
  <tabstops>
    <tabstop>encryptAllPartsRB</tabstop>
    <tabstop>askEachPartRB</tabstop>
    <tabstop>dontEncryptRB</tabstop>
    <tabstop>warnUnencryptedCB</tabstop>
    <tabstop>useCRLsCB</tabstop>
    <tabstop>warnCRLExpireCB</tabstop>
    <tabstop>warnCRLExpireSB</tabstop>
    <tabstop>encryptionAlgorithmCO</tabstop>
    <tabstop>warnReceiverCertificateExpiresCB</tabstop>
    <tabstop>warnReceiverCertificateExpiresSB</tabstop>
    <tabstop>warnChainCertificateExpiresCB</tabstop>
    <tabstop>warnChainCertificateExpiresSB</tabstop>
    <tabstop>warnReceiverNotInCertificateCB</tabstop>
    <tabstop>storeEncryptedCB</tabstop>
    <tabstop>checkCertificatePathCB</tabstop>
    <tabstop>alwaysCheckRootRB</tabstop>
    <tabstop>pathMayEndLocallyCB</tabstop>
  </tabstops>
  <connections>
    <connection>
      <sender>warnReceiverCertificateExpiresCB</sender>
      <signal>toggled(bool)</signal>
      <receiver>warnReceiverCertificateExpiresSB</receiver>
      <slot>setEnabled(bool)</slot>
    </connection>
    <connection>
      <sender>warnChainCertificateExpiresCB</sender>
      <signal>toggled(bool)</signal>
      <receiver>warnChainCertificateExpiresSB</receiver>
      <slot>setEnabled(bool)</slot>
    </connection>
    <connection>
      <sender>checkCertificatePathCB</sender>
      <signal>toggled(bool)</signal>
      <receiver>alwaysCheckRootRB</receiver>
      <slot>setEnabled(bool)</slot>
    </connection>
    <connection>
      <sender>checkCertificatePathCB</sender>
      <signal>toggled(bool)</signal>
      <receiver>pathMayEndLocallyCB</receiver>
      <slot>setEnabled(bool)</slot>
    </connection>
    <connection>
      <sender>useCRLsCB</sender>
      <signal>toggled(bool)</signal>
      <receiver>warnCRLExpireCB</receiver>
      <slot>setEnabled(bool)</slot>
    </connection>
    <connection>
      <sender>useCRLsCB</sender>
      <signal>toggled(bool)</signal>
      <receiver>warnCRLExpireSB</receiver>
      <slot>setEnabled(bool)</slot>
    </connection>
    <connection>
      <sender>warnCRLExpireCB</sender>
      <signal>toggled(bool)</signal>
      <receiver>warnCRLExpireSB</receiver>
      <slot>setEnabled(bool)</slot>
    </connection>
  </connections>
</ui>
